### PR TITLE
feat: add player search functionality

### PR DESCRIPTION
**Pull request by crubio**
Build of the api client, rendering component and demo on the `<Home>` page. 

⚠️ This push also removes the dot env file, so that should be added locally by the user - this is easy enough to figure out what needs to be put in there.

The API accepts wild card origin requests now, so its possible to test this from more networked devices.

### CODE REVIEW

File: src/features/player-search/components/index.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  return (
+    <>
+      <Box sx={{ width: 500, maxWidth: '100%' }}>
+        <TextField 
+          fullWidth
+          label="player name"
+          id="player-name" 
+          margin="normal"
+          onChange={(event) => { handleSubmit(event.target.value); }}
+        />
```
Comment (pacifisticuffs) [inline src/features/player-search/components/index.tsx]: tiny improvement suggestion: make it `type="search"` and i think `autocomplete="off"`. that should render the handy little `X` thinger on mobile to clear out the contents in one click

**Review by pacifisticuffs — APPROVED** [review]
🚀

File: src/features/player-search/components/index.tsx (hunk 1 above)
Comment (crubio) [reply src/features/player-search/components/index.tsx]: oh hell yeah bb, will try that out.

File: src/pages/Home.tsx [hunk 2]
```diff
@@ -1,12 +1,51 @@
...
+  // Search query is made from the page level, not the component level
+  const {data} = useQuery<Array<Player>>({
+    enabled: queryEnabled,
+    queryKey: ['player-search', queryId ,searchQuery],
+    queryFn: async () => getPlayerSearch(searchQuery.playerName),
+  });
+
+  const handleSubmit = (playerName: string): void => {
+    if (!playerName || playerName.length < 3) {
+      setPlayerData([]);
```
Comment (crubio) [inline src/pages/Home.tsx:25]: Clear out the state of the data if we have less than the necessary characters for an API call.

**Pull request merged**